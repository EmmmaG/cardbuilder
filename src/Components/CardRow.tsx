import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const styles = (theme: Theme) => ({
    rowContainer: {
        position: 'relative' as 'relative',
        height: '35px',
    },
    titleContainer: {
        display: 'inline-block',
        borderBottom: `1px solid ${theme.palette.grey[300]}`,
        width: '80px',
        position: 'absolute' as 'absolute',
        bottom: 0,
        left: 0,
    },
    contentContainer: {
        display: 'inline-block',
        borderBottom: `1px solid ${theme.palette.grey[300]}`,
        position: 'absolute' as 'absolute',
        bottom: 0,
        left: '80px',
        width: '310px',
    },
    content: {
        whiteSpace: 'nowrap' as 'nowrap',
        overflowX: 'hidden' as 'hidden',
        textOverflow: 'ellipsis',
        fontFamily: '"Big Caslon", "Times New Roman", "serif"',
        fontSize: '18px',
        textAlign: 'left' as 'left',
    },
    title: {
        fontSize: '12px',
        color: '#374D60',
        textAlign: 'left' as 'left',
    },
})

export interface Props extends WithStyles<typeof styles> {
    title?: string
    content: string
    inline?: boolean
}

export class CardRow extends React.PureComponent<Props> {
	public render() {
        const { classes, title, content, inline } = this.props
        const containerInlineStyle = inline ? { width: '195px' } : undefined
        const contentInlineStyle = inline ? { width: '115px' } : undefined

		return (
			<div className={classes.rowContainer} style={containerInlineStyle}>
                <div className={classes.titleContainer}>
                    {title && <Typography className={classes.title}>{title}</Typography>}
                </div>
                <div className={classes.contentContainer} style={contentInlineStyle}>
                    {content && (
                        <Typography
                            className={classes.content}
                        >
                            {content}
                        </Typography>
                    )}
                </div>
            </div>
		)
	}
}

export default withStyles(styles)(CardRow)
