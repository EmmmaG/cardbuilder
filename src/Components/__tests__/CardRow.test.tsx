import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import CardRow, { Props } from '../CardRow'

const baseProps: Props = {
    content: 'Sydney',
	classes: idObj,
}

describe('CardRow', () => {
    const render = (props?: Partial<Props>) => shallow(<CardRow {...{ ...baseProps, ...props }} />)

	it('renders correctly', () => {
		const wrapper = render().dive()
		expect(wrapper).toMatchSnapshot()
    })

    it('renders with correct props', () => {
        const wrapper = render({title: 'Address', inline: true }).dive()
		expect(wrapper).toMatchSnapshot()
    })
})
