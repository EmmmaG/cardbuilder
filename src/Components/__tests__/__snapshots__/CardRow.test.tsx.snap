// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRow renders correctly 1`] = `
<div
  className="CardRow-rowContainer-1"
>
  <div
    className="CardRow-titleContainer-2"
  />
  <div
    className="CardRow-contentContainer-3"
  >
    <WithStyles(Typography)
      className="CardRow-content-4"
    >
      Sydney
    </WithStyles(Typography)>
  </div>
</div>
`;

exports[`CardRow renders with correct props 1`] = `
<div
  className="CardRow-rowContainer-1"
  style={
    Object {
      "width": "195px",
    }
  }
>
  <div
    className="CardRow-titleContainer-2"
  >
    <WithStyles(Typography)
      className="CardRow-title-5"
    >
      Address
    </WithStyles(Typography)>
  </div>
  <div
    className="CardRow-contentContainer-3"
    style={
      Object {
        "width": "115px",
      }
    }
  >
    <WithStyles(Typography)
      className="CardRow-content-4"
    >
      Sydney
    </WithStyles(Typography)>
  </div>
</div>
`;
